app.go

```diff
@@ -308,11 +335,46 @@ func New(opts ...Option) *App {
 
 	if err := app.executeInvokes(); err != nil {
 		app.err = err
+		var b bytes.Buffer
+
+		if dig.CanVisualizeError(err) {
+			dig.Visualize(app.container, &b, dig.VisualizeError(err))
+		}
+
+		app.errorHooks.onError(wrappedErr{
+			graph: b.String(),
+			err:   err,
+		})
 	}
 
 	return app
 }
 
+type wrappedErr struct {
+	graph string
+	err   error
+}
+
+type isWrapped interface {
+	Graph() string
+}
+
+func (we wrappedErr) Graph() string {
+	return we.graph
+}
+
+func (we wrappedErr) Error() string {
+	return we.err.Error()
+}
+
+// VisualizeError returns the visualization of the error if available.
+func VisualizeError(err error) string {
+	if e, ok := err.(isWrapped); ok {
+		return e.Graph()
+	}
+	return ""
```

Comment:
I'd think that as a user of this library, if the error is not visualizable, I'd like some indication rather than an empty string -- however, I'm not entirely sure of the precedents set by the rest of `Fx`.

Maybe this function should also return an error, or at least a standard message like:
`"cannot visualize error: ..."`

Reply:
Good call. I've changed the function to return an error if there is no graph or if the graph is empty. That actually eliminates some of the checks needed when I'm calling it elsewhere.